src/hotspot/share/gc/g1/g1BarrierSet.cpp

```diff
@@ -49,11 +49,20 @@
 
 class G1BarrierSetC1;
 class G1BarrierSetC2;
+#if G1_LATE_BARRIER_MIGRATION_SUPPORT
+class G1BarrierSetC2Early;
+#endif
 
 G1BarrierSet::G1BarrierSet(G1CardTable* card_table) :
   CardTableBarrierSet(make_barrier_set_assembler<G1BarrierSetAssembler>(),
                       make_barrier_set_c1<G1BarrierSetC1>(),
+#if G1_LATE_BARRIER_MIGRATION_SUPPORT
+                      G1UseLateBarrierExpansion ?
+                      make_barrier_set_c2<G1BarrierSetC2>() :
+                      make_barrier_set_c2<G1BarrierSetC2Early>(),
+#else
                       make_barrier_set_c2<G1BarrierSetC2>(),
+#endif
```

Comment:
I assume it is temporary until all ports a ready (except 32-bit x86 may be). Right?

Reply:
Right, all code guarded by `G1_LATE_BARRIER_MIGRATION_SUPPORT` will be removed before integration.